x/ccv/provider/keeper/hooks.go

```diff
@@ -167,3 +167,7 @@ func (h Hooks) AfterValidatorBeginUnbonding(_ sdk.Context, _ sdk.ConsAddress, _
 func (h Hooks) BeforeDelegationRemoved(_ sdk.Context, _ sdk.AccAddress, _ sdk.ValAddress) error {
 	return nil
 }
+
+func (h Hooks) BeforeTokenizeShareRecordRemoved(_ sdk.Context, _ uint64) error {
```

Comment:
This is not mentioned anywhere else in the code base - I saw the same is true for some other hooks, too.
Just double-checking and for my understanding: is this for compatibility with something?

Reply:
Hooks are needed to support the staking keeper interface. Most of them are no-ops.

I think we override them to be sure that we use the method specific to the `providerKeeper` since we cannot control what the `sdk.StakingKeeper` hooks do.